src/test/java/seedu/address/logic/commands/FindGoodCommandTest.java

```diff
@@ -0,0 +1,96 @@
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.good.GoodSupplierPairContainsKeywordsPredicate;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code FindGoodCommand}.
+ */
+public class FindGoodCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalInventory(),
+            getTypicalTransactionHistory(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), getTypicalInventory(),
+            getTypicalTransactionHistory(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        GoodSupplierPairContainsKeywordsPredicate firstPredicate =
+                new GoodSupplierPairContainsKeywordsPredicate(Collections.singletonList("apple"));
+        GoodSupplierPairContainsKeywordsPredicate secondPredicate =
+                new GoodSupplierPairContainsKeywordsPredicate(Collections.singletonList("banana"));
+
+        FindGoodCommand findFirstCommand = new FindGoodCommand(firstPredicate);
+        FindGoodCommand findSecondCommand = new FindGoodCommand(secondPredicate);
+
+        // same object -> returns true
+        assertTrue(findFirstCommand.equals(findFirstCommand));
+
+        // same values -> returns true
+        FindGoodCommand findFirstCommandCopy = new FindGoodCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different supplier -> returns false
+        assertFalse(findFirstCommand.equals(findSecondCommand));
+    }
+
+    @Test
+    public void execute_zeroKeywords_noGoodFound() {
+        String expectedMessage = String.format(MESSAGE_SUPPLIERS_LISTED_OVERVIEW, 0);
+        GoodSupplierPairContainsKeywordsPredicate predicate = preparePredicate(" ");
+        FindGoodCommand command = new FindGoodCommand(predicate);
+        expectedModel.updateFilteredSupplierList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredSupplierList());
+    }
+
+    @Test
+    public void execute_multipleKeywords_multipleSuppliersFound() {
+        String expectedMessage = String.format(MESSAGE_SUPPLIERS_LISTED_OVERVIEW, 3);
+        GoodSupplierPairContainsKeywordsPredicate predicate = preparePredicate("hand pear paper");
+        FindGoodCommand command = new FindGoodCommand(predicate);
+        expectedModel.updateFilteredSupplierList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(ALICE, BENSON, DANIEL), model.getFilteredSupplierList());
```

Comment:
I think this test case is very dependent on the typical list contents. If someone changes the list of typical items, it will affect this test. Is it possible to make a custom supplier list with their own offers?